src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx

```diff
@@ -84,124 +83,149 @@ const ShortCircuitAnalysisResultTable: FunctionComponent<
 > = ({
     result,
     analysisType,
-    updateFilter,
-    updateSort,
     isFetching,
-    faultTypeOptions,
-    limitViolationTypeOptions,
+    sortProps,
+    filterProps,
+    filterEnums,
 }) => {
     const intl = useIntl();
     const theme = useTheme();
 
-    const textFilterParams = useMemo(() => {
-        return {
-            debounceMs: 1200, // we don't want to fetch the back end too fast
-            maxNumConditions: 1,
-            filterOptions: ['contains', 'startsWith'],
-            textMatcher: (): boolean => true, // we disable the AGGrid filter because we do it in the server
-        };
-    }, []);
-
     const columns = useMemo(
         () => [
-            {
+            makeAgGridCustomHeaderColumn({
                 headerName: intl.formatMessage({ id: 'IDNode' }),
                 field: 'elementId',
-                sortable: analysisType === ShortCircuitAnalysisType.ALL_BUSES,
-                filter:
+                sortProps:
                     analysisType === ShortCircuitAnalysisType.ALL_BUSES
-                        ? 'agTextColumnFilter'
-                        : null,
-                filterParams: textFilterParams,
-            },
-            {
+                        ? sortProps
+                        : undefined,
```

Comment:
You can create two constants, for example, sortPropsChecked and filterPropsChecked, to avoid having to perform the verification for each column